From 1de860062220597308d45bcbf6bcbab324c4363c Mon Sep 17 00:00:00 2001
From: Neena Busireddy <neenareddy.busireddy@nxp.com>
Date: Mon, 17 Dec 2018 10:02:19 -0600
Subject: [PATCH] units/systemd-udevd: Set PrivateMounts to no

Fix automount problem in systemd 239 version.

In systemd 239 version, A new unit file setting PrivateMounts= has been added.
It's a boolean option. If enabled the unit's processes are invoked in their own file
system namespace.

Set PrivateMounts=no to ensure that mounts and unmounts are propagated from systemd's
namespace to the service's namespace from systemd 239.

Also, SystemCallFilter=@system-service @module @raw-io is added in systemd 239. The simplest
way to regain the previous behaviour is to remove it entirely.

https://unix.stackexchange.com/questions/464959/udev-rule-to-automount-media-devices-stopped-working-after-systemd-was-updated-t

Upstream-Status: Inappropriate [i.MX specific]

Signed-off-by: Neena Busireddy <neenareddy.busireddy@nxp.com>

---
 units/systemd-udevd.service.in | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/units/systemd-udevd.service.in b/units/systemd-udevd.service.in
index dfc2a0e341..d2e0e0d069 100644
--- a/units/systemd-udevd.service.in
+++ b/units/systemd-udevd.service.in
@@ -27,13 +27,12 @@ RestartSec=0
 ExecStart={{ROOTLIBEXECDIR}}/systemd-udevd
 KillMode=mixed
 TasksMax=infinity
-PrivateMounts=yes
+PrivateMounts=no
 ProtectHostname=yes
 MemoryDenyWriteExecute=yes
 RestrictAddressFamilies=AF_UNIX AF_NETLINK AF_INET AF_INET6
 RestrictRealtime=yes
 RestrictSUIDSGID=yes
-SystemCallFilter=@system-service @module @raw-io bpf
 SystemCallFilter=~@clock
 SystemCallErrorNumber=EPERM
 SystemCallArchitectures=native
